docs/vec_of_option.md

```diff
@@ -0,0 +1,266 @@
+
+        for i in 0..first {
+            unsafe { validity.set_unchecked(i, false) };
+        }
+
+        for i in last..chunked_arr.len() {
+            unsafe { validity.set_unchecked(i, false) };
+            out.push(Zero::zero())
+        }
+
+        let array = PrimitiveArray::new(
+            T::get_dtype().to_arrow(CompatLevel::newest()),
+            out.into(),
+            Some(validity.into()),
+        );
+        ChunkedArray::with_chunk(chunked_arr.name(), array)
+    } else {
+        ChunkedArray::from_vec(chunked_arr.name(), out)
+    }
+}
+```
+
+That's a lot to digest at once, so let's take small steps and focus on the core logic.
+At the start, we store the indexes of the first and last null values:
+
+```rust
+let first = chunked_arr.first_non_null().unwrap();
+let last = chunked_arr.last_non_null().unwrap() + 1;
+```
+
+Then, we push actual zeroes to the output `Vec` - as many as there are nulls before the first valid value in the input:
+
+```rust
+let mut out = Vec::with_capacity(chunked_arr.len());
+for _ in 0..first {
+    out.push(Zero::zero());
+}
+```
+
+We skip the first `first` elements and start interpolating, but the droids we're looking for are not here, so let's skip that part:
+
+```rust
+let mut iter = chunked_arr.iter().skip(first);
+let mut low = iter.next().unwrap().unwrap();
+out.push(low);
+while let Some(next) = iter.next() {
+    // Interpolation logic
+}
+```
+
+Now, after _most_ of the work is done, we find something interesting - see the comments:
+
+```rust
+if first != 0 || last != chunked_arr.len() {
+    // A validity mask is created for the vector, initially all set to true
+    let mut validity = MutableBitmap::with_capacity(chunked_arr.len());
+    validity.extend_constant(chunked_arr.len(), true);
+
+    for i in 0..first {
+        // The indexes corresponding to the zeroes before the first valid value
+        // are set to false (invalid)
+        unsafe { validity.set_unchecked(i, false) };
+    }
+
+    for i in last..chunked_arr.len() {
+        // The indexes corresponding to the values after the last valid value
+        // are set to false (invalid)
+        unsafe { validity.set_unchecked(i, false) };
+
+        out.push(Zero::zero())  // This is equivalent to the zeroes pushed
+                                // before the first valid value, it's just done
+                                // out of order
+    }
+
+    let array = PrimitiveArray::new(
+        T::get_dtype().to_arrow(CompatLevel::newest()),
+        out.into(),
+        Some(validity.into()),
+    );
+    ChunkedArray::with_chunk(chunked_arr.name(), array)
+} else {
+    ChunkedArray::from_vec(chunked_arr.name(), out)
+}
+```
+
+Notice how only the outer nulls were set to invalid (if the inner nulls were as well, this wouldn't be a very good interpolation).
+The validity mask already exists for all Chunked types, so we could dismiss its overhead in this scenario.  
```

Comment:
this isn't quite accurate - if a chunkedarray is created without missing values, then the validity is just `None` (which is why, in the code above, the validity is only created if `first` / `last` are such that it's required

Reply:
Thanks for catching that!